include/PetriEngine/Reachability/ReachabilitySearch.h

```diff
@@ -196,6 +216,93 @@ namespace PetriEngine {
             _max_tokens = states.maxTokens();
             return false;
         }
+
+        template<typename G>
+        bool ReachabilitySearch::tryReachRandomWalk(std::vector<std::shared_ptr<PQL::Condition> >& queries,
+                                                    std::vector<ResultPrinter::Result>& results, bool usequeries,
+                                                    StatisticsLevel statisticsLevel, size_t seed,
+                                                    int depthRandomWalk, const int incRandomWalk)
+        {
+            // Set up state
+            searchstate_t ss;
+            ss.enabledTransitionsCount.resize(_net.numberOfTransitions(), 0);
+            ss.expandedStates = 0;
+            ss.exploredStates = 1;
+            ss.heurquery = queries.size() >= 2 ? std::rand() % queries.size() : 0;
+            ss.usequeries = usequeries;
+            const PQL::Condition *query = queries[ss.heurquery].get();
+
+            // Set up working area
+            _initial.setMarking(_net.makeInitialMarking());
+            Structures::State candidate; // Candidate for the next step, modified by the generator
+            Structures::State currentStepState;
+            candidate.setMarking(_net.makeInitialMarking());
+            currentStepState.setMarking(_net.makeInitialMarking());
+
+            Structures::RandomWalkStateSet states(_net, _kbound, query, seed); // State set
+            G generator = _makeSucGen<G>(_net, queries);                       // Successor generator
+
+            // Check initial marking
+            if(ss.usequeries)
+            {
+                if(checkQueries(queries, results, _initial, ss, &states))
+                {
+                    if(statisticsLevel != StatisticsLevel::None)
+                        printStats(ss, &states, statisticsLevel);
+                    _max_tokens = states.maxTokens();
+                    return true;
+                }
+            }
+
+            const int maxDepthValue = std::numeric_limits<int>::max() - incRandomWalk;
```

Comment:
Don't use int, use a fixed width (int64_t or int32_t) depending on what is needed. Use int64_t if in doubt.

Reply:
Should I change depthRandomWalk into int64_t as well?

Reply:
Yes, I think that Peter's point is to use always int64_t because size of int may depend on the architecture etc.